ui/cypress/support/helpers.js

```diff
@@ -121,38 +150,43 @@ export async function waitUrlToProcess(url, property, maxTries = 6, method='GET'
 
 /**
  * NOTE Returns a regular promise, not a "cypress one".
-        Use cy.wrap(this_result) to convert to cypress promise on test.
+ *      Use cy.wrap(this_result) to convert to cypress promise on test.
  **/
-export async function provisionAndWaitReady(existingModelId) {
-  console.log('provisionAndWaitReady existingModelId', existingModelId);
-
+export async function provisionTerminalWaitReady(existingModelId) {
   const modelId = existingModelId;
 
-  Cypress.log({message: `Provisioning Terminal for model ${modelId}.`});
+  cy.log(`Provisioning Terminal for model ${modelId}.`);
 
-  const response = await p(cy.request('GET', '/api/terminal/docker/locks'));
+  const response = await axiosClient({
+    url: '/api/terminal/docker/locks',
+    auth
+ });
 
-  let cy_promise;
-  if (get(response, 'body.locks[0].modelId')) {
-    if (get(response, 'body.locks[0].modelId') !== modelId) {
-      cy_promise = shutdownWorker()
-        .then(() => {
-          return provisionModelTerminal(modelId);
-        });
+  let promise;
+  if (get(response, 'data.locks[0].modelId')) {
+    if (get(response, 'data.locks[0].modelId') !== modelId) {
+      await shutdownWorker();
+      promise = provisionModelTerminal(modelId);
```

Comment:
Refactored to use await since now all promises are regular promises (non-cypress promise). If this sounds confusing, read:
https://docs.cypress.io/api/utilities/promise